### DISCUSSION TITLE
What's keeping C401 (non N) from working?

### DISCUSSION THREAD

**Discussion by CReimer** [pvvx/ZigbeeTLc#121] [General]
I already have a bunch of Zigbee hardware and switching my C401s over to Zigbee could help me get rid of my BLE hardware.
Unfortunately I have a mix of C401 and C401N and only the N version is supported.

Looking at the README it seems to be planned. I'm curious what's the technical reason behind the old version not being supported. Is there a big difference between C401 and C401N?

Thanks

**Comment by pvvx** [comment]
There is a big difference in the type of E-Ink and the controller. The update time of the old versions is more than 1.5 seconds. All this time it is necessary to work with the display controller, which imposes an increase in consumption and a change in the program algorithms. Therefore, there are no software with slow displays for thermometers.

There is no sense in Zigbee. Ble in Le Long Range mode is long distances than Zigbee and WiFi.

![image](https://github.com/user-attachments/assets/0e6ef95a-950f-4bbe-a1e8-34ce63c75620)
At these reception levels Zigbee, Wifi, BLE 4.2 do not work:
These are distances over 500 meters at TX 0 dBm.

At the same time, BLE (LE Long Range) consumption is several times less than that of Zigbee.
![image](https://github.com/user-attachments/assets/17106b88-b31d-431a-8559-21ece03a6cc9)
[Xiaomi Mijia MJWSD05MMC](https://pvvx.github.io/MJWSD05MMC) (transmission every 5 seconds, measurement every 15 seconds).
